### PR TITLE
Provide an option to send cross-origin credentials.

### PR DESCRIPTION
**Pull request by robholland**
## What was changed
Added an option to send credentials to the remote codec server.

## Why?
Some users would like to make use of cookies for authentication to their remote codec server. However CORS would refuse requests that claim they'd like to send credentials if the user's remote codec server does not expect them. Make this an option for users who need it.

This code is based on #1050 with code added to make it optional.

Screen Shot 2023-03-27 at 3 28 03 PM

## Checklist

1. Closes 

2. How was this tested:

3. Any docs updates needed?

### CODE REVIEW

**Comment by Alex-Tideman** [comment]
@robholland Would we want to show this option for cloud.temporal.io?

**Comment by robholland** [comment]
Yes, it would be applicable to both cloud and OSS installs.

File: src/lib/services/settings-service.ts [hunk 1]
```diff
@@ -25,6 +25,7 @@ export const fetchSettings = async (request = fetch): Promise<Settings> => {
     codec: {
       endpoint: settingsResponse?.Codec?.Endpoint,
       passAccessToken: settingsResponse?.Codec?.PassAccessToken,
+      passCredentials: settingsResponse?.Codec?.PassCredentials,
```
Comment (Alex-Tideman) [inline src/lib/services/settings-service.ts:28]: Are we adding this to the settings api through an env var?

**Comment by Alex-Tideman** [comment]
I know Candace has updates to the modal itself, I'm going to run this new option by her and then update the modal in this PR.

File: src/lib/services/settings-service.ts (hunk 1 above)
Comment (robholland) [reply src/lib/services/settings-service.ts:28]: Yeah we'll want to do that like we do for the access token.

**Review by Alex-Tideman — APPROVED** [review]

**Pull request merged**